microcosm_pubsub/chain/chain.py

```diff
@@ -63,18 +63,28 @@ def __call__(self, context=None, **kwargs):
         context = context or self.new_context_type()
         context.update(kwargs)
 
-        # Allow self reference
-        if "context" not in context:
-            context.update(context=context)
-        elif context["context"] is not context:
-            raise ValueError("context should be set as context")
-
         for link in self.links:
-            res = self.apply_decorators(context, link)()
+            func = self.apply_decorators(context, link)
```

Comment:
Make the call chain more explicit here by separating the decoration and execution logic. I had a troublesome bug in my local context implementation that I mistakenly believed was part of the decorator logic because these steps were not separated.